trieste/models/keras/models.py

```diff
@@ -393,20 +450,38 @@ def optimize_encoded(self, dataset: Dataset) -> tf_keras.callbacks.History:
 
         :param dataset: The data with which to optimize the model.
         """
-        fit_args = dict(self.optimizer.fit_args)
+        fit_args_copy = copy.deepcopy(dict(self.optimizer.fit_args))
 
         # Tell optimizer how many epochs have been used before: the optimizer will "continue"
         # optimization across multiple BO iterations rather than start fresh at each iteration.
         # This allows us to monitor training across iterations.
-
-        if "epochs" in fit_args:
-            fit_args["epochs"] = fit_args["epochs"] + self._absolute_epochs
+        if "epochs" in fit_args_copy:
+            fit_args_copy["epochs"] = fit_args_copy["epochs"] + self._absolute_epochs
 
         x, y = self.prepare_dataset(dataset)
+
+        validation_split = fit_args_copy.pop("validation_split", 0.0)
+        tf_data = self.prepare_tf_data(
+            x,
+            y,
+            batch_size=fit_args_copy["batch_size"],
```

Comment:
"batch_size" isn't guaranteed to exist for a user-supplied fit_args

```suggestion
            batch_size=fit_args_copy.get("batch_size"),
```

Reply:
ah, well spotted, I was remembering BatchOptimizer...